src/main/java/org/codehaus/mojo/license/download/LicensedArtifactResolver.java

```diff
@@ -216,4 +231,8 @@ public void loadProjectDependencies( ResolvedProjectDependencies artifacts,
     }
     // CHECKSTYLE_ON: MethodLength
 
+    public void setAetherRepoSession( RepositorySystemSession aetherRepoSession )
+    {
+        this.aetherRepoSession = aetherRepoSession;
```

Comment:
Making aetherRepoSession mutable in this way does not feel good. If there is no way to persuade plexus to inject it, I vote for removing the aetherRepoSession field and pass it via method parameter.

Reply:
Replaced with MavenSession injection